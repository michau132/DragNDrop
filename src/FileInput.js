/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import styled from 'styled-components';
import { Typography, Button } from '@material-ui/core';
import PropTypes from 'prop-types';

const StyledTypography = styled(Typography)`
  && {
    margin-right: 20px;
  }
`;

class Input extends React.Component {
  static propTypes = {
    submitFile: PropTypes.func.isRequired,
  }

  constructor(props) {
    super(props);
    this.fileInput = React.createRef();
  }

  getFiles = () => {
    const { files } = this.fileInput.current;
    const { submitFile } = this.props;
    submitFile(files);
    this.fileInput.current.value = '';
  }

  render() {
    return (
      <div>
        <StyledTypography variant="subtitle1">Upload your files</StyledTypography>
        <input
          hidden
          id="file"
          type="file"
          name="files[]"
          accept="image/*"
          multiple
          ref={this.fileInput}
          onChange={this.getFiles}
        />
        <label htmlFor="file">
          <Button
            color="primary"
            variant="contained"
            component="span"
          >
          Upload
          </Button>
        </label>
      </div>
    );
  }
}

export default Input;
