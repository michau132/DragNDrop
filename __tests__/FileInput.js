import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FileInput from '../src/FileInput';

const props = {
  submitFile: jest.fn(),
};

describe('<FileInput/> component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<FileInput {...props} />);
  });
  test('should be defined', () => {
    expect(wrapper).toBeDefined();
  });
  test('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test('should call submit file', () => {
    wrapper.instance().fileInput.current = { files: { name: 'fileName' } };
    wrapper.find('input').simulate('change');
    expect(props.submitFile).toHaveBeenCalled();
  });
});
