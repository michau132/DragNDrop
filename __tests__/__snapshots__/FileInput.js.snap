// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileInput/> component should match snapshot 1`] = `
<div>
  <Styled(WithStyles(Typography))
    variant="subtitle1"
  >
    Upload your files
  </Styled(WithStyles(Typography))>
  <input
    accept="image/*"
    hidden={true}
    id="file"
    multiple={true}
    name="files[]"
    onChange={[Function]}
    type="file"
  />
  <label
    htmlFor="file"
  >
    <WithStyles(Button)
      color="primary"
      component="span"
      variant="contained"
    >
      Upload
    </WithStyles(Button)>
  </label>
</div>
`;
